src/components/FilesSideBar/FileMenuSideBar.tsx

```diff
@@ -1,42 +1,117 @@
-import { Typography } from "@mui/material";
-import FileMenuOption from "./FileMenuOption/FileMenuOption";
+import React, { useState } from "react";
+import {
+  Typography,
+  CssBaseline,
+  Drawer,
+  IconButton,
+  useMediaQuery,
+  Theme,
+} from "@mui/material";
+import MenuIcon from "@mui/icons-material/Menu";
+import DirectoryOption from "./FileMenuOption/DirectoryOption";
+import FileOption from "./FileMenuOption/FileOption";
 import styles from "./FileMenuSideBar.module.css";
-import { useState } from "react";
-import { splitUntilSecondSlash } from "../../utils/methods";
+import organizeFiles, { FileNode } from "./FileNode";
+
+const DRAWER_WIDTH = 300;
 
 interface FileMenuSideBarProps {
-  proyectFiles: string[];
-  optionOnClick: (val: string) => void;
+  projectFiles: string[];
+  basePath: string;
+  setSelectedFilePath: (e: string) => void;
 }
 
 const FileMenuSideBar: React.FC<FileMenuSideBarProps> = ({
-  proyectFiles,
-  optionOnClick,
+  projectFiles,
+  basePath,
+  setSelectedFilePath,
 }) => {
-  const [fileSelected, setFileSelected] = useState<string>(proyectFiles[0]);
+  const [fileSelected, setFileSelected] = useState<string>(projectFiles[0]);
+  const [mobileOpen, setMobileOpen] = useState(false);
+  const isLargeScreen = useMediaQuery((theme: Theme) =>
+    theme.breakpoints.up("md")
+  );
 
-  return (
+  const nodes: FileNode[] = organizeFiles(projectFiles.slice(1), basePath);
+
+  function onFileSelected(value: string, path: string): void {
+    setFileSelected(value);
+    setSelectedFilePath(path);
+  }
+
+  const handleDrawerToggle = () => {
+    setMobileOpen(!mobileOpen);
+  };
+
+  const DrawerContent = (
```

Comment:
Please move this component to another file, you have two components in the same file